src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1700,6 +1680,26 @@ bool canIgnoreAnonymousType(NamedTypeSymbol type)
                 return AnonymousTypeManager.GetAnonymousTypePropertyTypesWithAnnotations(type).Any(t => canIgnoreAnyType(t.Type));
             }
         }
+
+        private static bool AreCloseEnough(Symbol original, Symbol updated)
+        {
+            return (original, updated) switch
+            {
+                (LambdaSymbol l, NamedTypeSymbol n) _ when n.IsDelegateType() => areLambdaAndNewDelegateSimilar(l, n),
+                (FieldSymbol { IsTupleField: true, TupleElementIndex: var oi } originalField, FieldSymbol { IsTupleField: true, TupleElementIndex: var ui } updatedField) =>
+                    originalField.Type.Equals(updatedField.Type, TypeCompareKind.AllNullableIgnoreOptions | TypeCompareKind.IgnoreTupleNames) && oi == ui,
+                _ => original.Equals(updated, TypeCompareKind.AllNullableIgnoreOptions | TypeCompareKind.IgnoreTupleNames)
+            };
+
+            static bool areLambdaAndNewDelegateSimilar(LambdaSymbol l, NamedTypeSymbol n)
+            {
+                var invokeMethod = n.DelegateInvokeMethod;
+                return invokeMethod.Parameters.SequenceEqual(l.Parameters,
```

Comment:
>SequenceEqual [](start = 47, length = 13)

Count of parameters might not match in error scenarios.

Reply:
We don't have a test that fails this currently. Agreed that this can happen in error scenarios, but in that case we should probably drop the updated symbol, rather than record such a symbol, as that will observably change quickinfo and potentially IDE refactorings. I'll make a change to check just the lambda section when adding to the map, and abort if the lambda is different by parameter count.

---
In reply to: [329732324](https://github.com/dotnet/roslyn/pull/38769#discussion_r329732324) [](ancestors = 329732324)